test/server/server.js

```diff
@@ -0,0 +1,228 @@
+  formatDate
+} from '../../src/server/server';
+
+export const padSingleDigitWithZeroTest = () => {
+  describe('padSingleDigitWithZero', () => {
+    it('should pad 0 with a 0 and stringify', () => {
+      expect(padSingleDigitWithZero(0))
+        .to
+        .equal('00');
+    });
+    it('should pad \'1\' with a 0', () => {
+      expect(padSingleDigitWithZero('1'))
+        .to
+        .equal('01');
+    });
+    it('should pad 2 with a 0 and stringify', () => {
+      expect(padSingleDigitWithZero(2))
+        .to
+        .equal('02');
+    });
+    it('should pad \'3\' with a 0', () => {
+      expect(padSingleDigitWithZero('3'))
+        .to
+        .equal('03');
+    });
+    it('should pad 4 with a 0 and stringify', () => {
+      expect(padSingleDigitWithZero(4))
+        .to
+        .equal('04');
+    });
+    it('should pad \'5\' with a 0', () => {
+      expect(padSingleDigitWithZero('5'))
+        .to
+        .equal('05');
+    });
+    it('should pad 6 with a 0 and stringify', () => {
+      expect(padSingleDigitWithZero(6))
+        .to
+        .equal('06');
+    });
+    it('should pad \'7\' with a 0', () => {
+      expect(padSingleDigitWithZero('7'))
+        .to
+        .equal('07');
+    });
+    it('should pad 8 with a 0 and stringify', () => {
+      expect(padSingleDigitWithZero(8))
+        .to
+        .equal('08');
+    });
+    it('should pad \'9\' with a 0', () => {
+      expect(padSingleDigitWithZero('9'))
+        .to
+        .equal('09');
+    });
+    it('should not change 10 but stringify', () => {
+      expect(padSingleDigitWithZero(10))
+        .to
+        .equal('10');
+    });
+    it('should not change \'11\'', () => {
+      expect(padSingleDigitWithZero('11'))
+        .to
+        .equal('11');
+    });
+    it('should not change 31', () => {
+      expect(padSingleDigitWithZero(31))
+        .to
+        .equal('31');
+    });
+    it('should throw an error if given an object', () => {
+      expect(() => {
+        padSingleDigitWithZero({});
+      }).to
+        .throw();
+    });
+    it('should throw an error if given an array', () => {
+      expect(() => {
+        padSingleDigitWithZero([]);
+      }).to
+        .throw();
+    });
+    it('should throw an error if given a boolean', () => {
+      expect(() => {
+        padSingleDigitWithZero(true);
+      }).to
+        .throw();
+    });
+    it('should throw an error if given null', () => {
+      expect(() => {
+        padSingleDigitWithZero(null);
+      }).to
+        .throw();
+    });
+  });
+};
+
+export const formatDateTest = () => {
+  const daysInMonth = {
+    january: 31,
+    february: 28,
+    march: 31,
+    april: 30,
+    may: 31,
+    june: 30,
+    july: 31,
+    august: 31,
+    september: 30,
+    october: 31,
+    november: 30,
+    december: 31
+  };
+  const YEAR = 2018;
+  describe('formatDate', () => {
+    it('should properly format January date on last day of month', () => {
```

Comment:
Again there is redundancy; testing the last day of the month on every month is not necessary and not super practical (what if there were 999999999 months?).  More valuable would be to test the last day of the month once (or maybe 3 times: for Feb, for one 30-day month, and for one 31-day month), the first day of the month once, and any other edge cases you can think of.  Maybe test the first and last month.  

And you should do one test that covers all the non-edge cases, e.g. just check a day in the middle of March.

From what I remember, you want a single test per thing that could go wrong, where only that one potential error is the focus of the test.  Don't double up here, e.g. if I think the last day of the month and the last month are both potential problems, I should write 2 tests, not one test where I just see if it works for the last day of the last month.  I should also write a minimum number of tests to cover the normal cases (as described in the previous paragraph).

Reply:
Edit: see my comment on the formatDate function. We should actually be using the dateFormat library instead of writing and testing a formatDate function